src/bootstrap-table.js

```diff
@@ -1567,7 +1567,7 @@ class BootstrapTable {
           this.options.undefinedText : value
       }
 
-      if (this.searchText !== '' && this.options.searchHighlight) {
+      if (this.searchText !== undefined && this.searchText !== '' && this.options.searchHighlight) {
```

Comment:
I think we could validate `searchText` as a `falsy` value then.

Reply:
I'm not sure what do you mean, could you please explain that further ?

Reply:
I agree on @UtechtDustin's fix. It's the simplest fix that addresses the **undefined** of **this.searchText**. When bootstrap-table is constructed, line [30](https://github.com/wenzhixin/bootstrap-table/blob/13ebc7e4a5bb95f8c6fbf0a1fc792ea71080f5b5/src/bootstrap-table.js#L30) is called, but it's the line 31 that initializes this.searchText.

Reply:
I mean if `searchText` can be `empty/null/undefined` we could validate it as `falsy` so:

`if (!this.searchText) {.....`

Reply:
Could this.searchText be set 0 if add **searchText** option in @UtechtDustin's example ?

$.getJSON('json/data1.json', function(tableData){
    $('#table').bootstrapTable({
   search: true,
   searchHighlight: true,
   searchText: 0,
   data: tableData
  });
 });

Reply:
@djhvscf @hs749 makes sense, i will change that later!